unsafe/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -74,6 +80,12 @@
    */
   private static final long PAGE_SIZE_BYTES = 1L << 26; // 64 megabytes
 
+  /**
+   * The maximum number of keys that BytesToBytesMap supports.
+   */
+  @VisibleForTesting
+  static final int MAX_CAPACITY = (1 << 30);
```

Comment:
The hashtable has to be power-of-2-sized.  Our long array can have at most (1 << 30) elements, since that's the largest power-of-2 that's less than Integer.MAX_VALUE, but we need two long array entries per record, so that gives us a maximum capacity of (1 << 29).